x-pack/plugins/session_view/public/components/process_tree_alerts/index.tsx

```diff
@@ -5,90 +5,50 @@
+import { ProcessTreeAlert } from '../process_tree_alert';
+import { MOUSE_EVENT_PLACEHOLDER } from '../../../common/constants';
 
 interface ProcessTreeAlertsDeps {
   alerts: ProcessEvent[];
+  jumpToAlertID?: string;
+  isProcessSelected?: boolean;
+  onAlertSelected: (e: MouseEvent) => void;
 }
 
-const getRuleUrl = (alert: ProcessEvent, http: CoreStart['http']) => {
-  return http.basePath.prepend(`/app/security/rules/id/${alert.kibana?.alert.rule.uuid}`);
-};
-
-const ProcessTreeAlert = ({ alert }: { alert: ProcessEvent }) => {
-  const { http } = useKibana<CoreStart>().services;
-
-  if (!alert.kibana) {
-    return null;
-  }
-
-  const { uuid, rule, original_event: event, workflow_status: status } = alert.kibana.alert;
-  const { name, query, severity } = rule;
-
-  return (
-    <EuiText key={uuid} size="s" data-test-subj={`sessionView:sessionViewAlertDetail-${uuid}`}>
-      <EuiFlexGroup>
-        <EuiFlexItem>
-          <h6>
-            <FormattedMessage id="xpack.sessionView.rule" defaultMessage="Rule" />
-          </h6>
-          {name}
-          <h6>
-            <FormattedMessage id="xpack.sessionView.query" defaultMessage="Query" />
-          </h6>
-          {query}
-        </EuiFlexItem>
-        <EuiFlexItem>
-          <h6>
-            <FormattedMessage id="xpack.sessionView.severity" defaultMessage="Severity" />
-          </h6>
-          {severity}
-          <h6>
-            <FormattedMessage
-              id="xpack.sessionView.workflowStatus"
-              defaultMessage="Workflow status"
-            />
-          </h6>
-          {status}
-        </EuiFlexItem>
-        <EuiFlexItem>
-          <h6>
-            <FormattedMessage id="xpack.sessionView.action" defaultMessage="Action" />
-          </h6>
-          {event.action}
-          <EuiSpacer />
-          <div>
-            <EuiButton
-              size="s"
-              href={getRuleUrl(alert, http)}
-              data-test-subj={`sessionView:sessionViewAlertDetailViewRule-${uuid}`}
-            >
-              <FormattedMessage id="xpack.sessionView.viewRule" defaultMessage="View rule" />
-            </EuiButton>
-          </div>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiText>
-  );
-};
-
-export function ProcessTreeAlerts({ alerts }: ProcessTreeAlertsDeps) {
+export function ProcessTreeAlerts({
+  alerts,
+  jumpToAlertID,
+  isProcessSelected = false,
+  onAlertSelected,
+}: ProcessTreeAlertsDeps) {
   const styles = useStyles();
+  const [selectedAlert, setSelectedAlert] = useState<ProcessEventAlert | null>(null);
 
   if (alerts.length === 0) {
     return null;
   }
 
+  const handleAlertClick = (alert: ProcessEventAlert | null) => {
+    onAlertSelected(MOUSE_EVENT_PLACEHOLDER);
+    setSelectedAlert(alert);
+  };
+
   return (
     <div css={styles.container} data-test-subj="sessionView:sessionViewAlertDetails">
       {alerts.map((alert: ProcessEvent) => (
```

Comment:
what about the following?

```js
{alerts.map((alert: ProcessEvent) => {

const alertUuid = alert.kibana?.alert.uuid || null;

return (
    <ProcessTreeAlert
          alert={alert}
          isInvestigated={jumpToAlertID === alertUuid}
          isSelected={isProcessSelected && selectedAlert?.uuid === alertUuid}
          onClick={handleAlertClick}
    />
)}
```

fall backing as null prevents extra checks like `!!jumpToAlertID && jumpToAlertID` because `undefined` won't match with `null`